python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -733,6 +734,10 @@ def sample_neighbors(
             corresponding to each neighboring edge of a node. It must be a 1D
             floating-point or boolean tensor, with the number of elements
             equalling the total number of edges.
+        returning_indices_is_optional: bool
```

Comment:
This is a very confusing option. We definitely need to rename it, and can you help me understand why do we need this option?

Reply:
Why is it confusing? One of the sampling results is the sampled `indices`. This option, when enabled makes it optional to return the sampled indices. I named it intentionally like this and thought it would be crystal clear.

Reply:
When it is enabled, returning indices becomes optional. Then, when it is not returned, it is our job to fetch it.

Reply:
For neighbor sampling, we need to fetch indices only for the sampled indices. Our current overlap_graph_fetch code path fetches the full insubgraph and overlap_graph_fetch was not providing speedup for NS in our regression results. This new code path provides speedup for NS.

Reply:
For an api, it is confusing to has a field says, it is optional for it to do a thing.
In the future, when someone try to debug / refactor the code, they will be confused on how to respect the argument, should I return or not return the indices.

Why not make it deterministic? If true, always return, otherwise never return?